Migration from LBO to manual deployment

ignasi35 opened the pull request:
Moving away from LBO: produce reference guide content and link from the Migration15.md 
  - [x] cluster bootstrap mechanisms
  - [x] health checks
  - [x] Lagom 1.5 won't work with LBO
  - [x] inter-service discovery (akka-discovery-service-locator)
  - [x] secret management

ihostage commented:
@ignasi35 Lightbend Orchestration like ConductR is deprecated? Lightbend is not planned to integrate LBO with Lagom in next releases?

ignasi35 commented:
> Lightbend is not planned to integrate LBO with Lagom in next releases?

Yes and no.

For example, `rp` (the CLI) will be discontinued but other details (akka management, service location, etc...) will be either enabled by default in Lagom or provided as opt-in dependencies. See all the details on the PR. ;-)

dwijnand commented on docs/manual/java/releases/Migration15.md:
```diff
@@ -46,30 +46,78 @@ were never intended for public consumption, and therefore have been marked depre
+In particular, ConductR tooling and Lightbend Orchestration handled some or all of the following:
 
-A new mechanism to form and [[join an Akka Cluster|Cluster#Joining]] is introduced in Lagom 1.5. Apart from the original `Manual Cluster Formation` the new `Akka Cluster Bootstrap` is now supported. This new mechanism is introduced with lower precedence than `Manual Cluster Formation` so if you rely on the use of a list of `seed-nodes` then everything will work as before. On the other hand, `Akka ...
+1. extend the your application with: cluster bootstrapping, akka management and health checks
```
the your

ignasi35 replied:
https://github.blog/changelog/2018-10-16-suggested-changes/ 😛

dwijnand replied:
I wasn't sure about the sentence on this one, so I just pointed out the typo.

dwijnand commented on docs/manual/java/releases/Migration15.md:
```diff
@@ -46,30 +46,78 @@ were never intended for public consumption, and therefore have been marked depre
+1. extend the your application with: cluster bootstrapping, akka management and health checks
+2. Service Location
+3. setup and produce docker images
+4. prepare the deployment specs for the target orchestrator
```
s/prepare/preparing/

You start the sentence with "ConductR and LBO handled some or all of the following" so you have to continue using the past tense for all the elements ("setting up and producing docker images", "extending ..." etc)

ignasi35 replied:
So instead of:

```
In particular, ConductR tooling and Lightbend Orchestration handled some or all of the following:

1. extend the application with: cluster bootstrapping, akka management and health checks
2. Service Location
3. setup and produce docker images
4. prepare the deployment specs for the target orchestrator
5. Secrets
```

I should use:

```
In particular, ConductR tooling and Lightbend Orchestration handled some or all of the following:

1. extending the application with: cluster bootstrapping, akka management and health checks
2. providing a Service Location implementation
3. setting up and producing docker images
4. preparing the deployment specs for the target orchestrator
5. accessing Secrets
```
?

dwijnand replied:
Yes, that's fine.  You can also keep "Service Location" and "Secrets" as nouns.

dwijnand commented on docs/manual/java/releases/Migration15.md:
```diff
@@ -46,30 +46,78 @@ were never intended for public consumption, and therefore have been marked depre
+
+#### Application extensions
+
+Starting with Lagom 1.5 your application will include [[Akka management HTTP|Cluster#Akka-Management]] out of the box with [[health checks|Cluster#Health-Checks]] enabled by default. Cluster formation also supports [[Cluster Bootstrapping|Cluster#Joining-during-production-(Akka-Cluster-Bootstrap)]] as a new way to form a cluster. These new defaults may require at least two changes on your codebas ...
```
> Starting with Lagom 1.5 your application will include [[Akka management HTTP|Cluster#Akka-Management]] out of the box with [[health checks|Cluster#Health-Checks]] enabled by default.

Maybe explain that Akka Management HTTP is included _in order_ to define health checks for the cluster.  Otherwise it remains unclear why it's added (IMO).

dwijnand commented on docs/manual/java/releases/Migration15.md:
```diff
@@ -46,30 +46,78 @@ were never intended for public consumption, and therefore have been marked depre
+
+#### Service Location
+
+You will have to add a `ServiceLocator `of your choice. We recommend using the new [`lagom-akka-discovery-service-locator`](https://github.com/lagom/lagom-akka-discovery-service-locator) which is implemented using [Akka Service Discovery](https://doc.akka.io/docs/akka/current/discovery/index.html) implementations.
```
```suggestion
As of Lagom 1.5 you will have to specify a Lagom `ServiceLocator` implementation to use.  The recommended choice is the one we developed on top of [Akka Service Discovery](https://doc.akka.io/docs/akka/current/discovery/index.html) which is therefore called [`lagom-akka-discovery-service-locator`](https://github.com/lagom/lagom-akka-discovery-service-locator).
```

ignasi35 replied:
Users always had to provide a `ServiceLocator`implementation so both my original writing and the suggestion are misleading. I'll rephrase.

dwijnand commented on docs/manual/java/releases/Migration15.md:
```diff
@@ -46,30 +46,78 @@ were never intended for public consumption, and therefore have been marked depre
+
+#### Docker images and deployment specs
+
+Regarding docker images and deployment specs, once you remove external tooling you will have to setup and maintain them manually. Instead of producing the `Dockerfile`, deployment scripts produced by tooling and orchestration specs from scratch, use the tooling to create these files and add them to git. You can later review and maintain them at will.
```
```suggestion
With the removal of ConductR or Lightbend Orchestration, the docker images and deployment specs will have to be maintained manually. Therefore the recommended migration is to take ownership of the `Dockerfile`, deployment scripts and orchestration specs created by such tooling, such as by committing them to the source repository and then maintaining them.  We also found that such maintenance can be made easier by making use of [kustomize](https://github.com/kubernetes-sigs/kustomize).
```

ignasi35 replied:
Oh, that's a lot better. Thanks! :-)

ignasi35 commented:
Thanks @dwijnand, I think I've addressed all comments.

I mentioned using `GH suggestions` but having the docs duplicate in `java` and `scala` makes suggestions a bit useless.

dwijnand commented:
Indeed.  We should solve that...

ignasi35 commented:
> We should solve that...

(cough) paradox include (/cough)

ignasi35 commented:
backported `1.5.x` 6609dc5d65c49d5eff0df2ef3178f056fddf9bd5
